### PR TITLE
Feature/token allowlist

### PR DESCRIPTION
**Pull request by piyalbasu**
Screenshot 2024-02-14 at 2 14 01 PM

### CODE REVIEW

File: extension/src/popup/components/manageAssets/ManageAssetRows/index.tsx [hunk 1]
```diff
@@ -279,78 +314,122 @@ export const ManageAssetRows = ({
...
+                          handleRowClick(
+                            { code, issuer, image, domain },
+                            isTrustlineActive,
+                          );
+                        }
+                      }}
+                      type="button"
+                      data-testid="ManageAssetRowButton"
+                    >
+                      {isTrustlineActive ? (
```
Comment (aristidesstaffieri) [inline extension/src/popup/components/manageAssets/ManageAssetRows/index.tsx:380]: Should this still be `isTrustlineActive || isContract`?

File: extension/src/popup/components/WarningMessages/index.tsx [hunk 2]
```diff
@@ -666,6 +670,138 @@ export const NewAssetWarning = ({
...
+            <div className="UnverifiedTokenWarning__icon">
+              <SorobanTokenIcon code={code} noMargin />
+            </div>
+            <div className="UnverifiedTokenWarning__code">{code}</div>
+            <div className="UnverifiedTokenWarning__domain">{domain}</div>
+            <div className="UnverifiedTokenWarning__description">
+              <div className="UnverifiedTokenWarning__description__icon">
+                <Icon.VerifiedUser />
+              </div>
+              <div className="UnverifiedTokenWarning__description__text">
```
Comment (aristidesstaffieri) [inline extension/src/popup/components/WarningMessages/index.tsx:743]: Would we show this even when adding a Soroban token? My thought here is that they don't have trustlines so maybe this inly belongs on classic assets or SACs?

**Review by aristidesstaffieri — APPROVED** [review]

File: extension/src/popup/components/manageAssets/ManageAssetRows/index.tsx (hunk 1 above)
Comment (piyalbasu) [reply extension/src/popup/components/manageAssets/ManageAssetRows/index.tsx:380]: Ah, good callout. I actually refactored `isTrustlineActive` up above to include all assets in `balances` (classic or token) and `tokensWithNoBalance`, so this should cover all asset/tokens a user has associated with their account. (at least in my testing it does 😬 )

File: extension/src/popup/components/WarningMessages/index.tsx (hunk 2 above)
Comment (piyalbasu) [reply extension/src/popup/components/WarningMessages/index.tsx:743]: Yup, we would show this if a user is adding a contract ID that doesn't appear on this list: 
https://api.stellar.expert/explorer/testnet/asset-list/top50

So we just flag that a user is adding a contract ID that has not been vetted

Lmk if that answers your question!

Comment (aristidesstaffieri) [reply extension/src/popup/components/WarningMessages/index.tsx:743]: Sounds good. I was mainly curious if the badge could be confused with the concept of having a trustline to a classic asset. As in this case we're "trusting" the asset by means of it being on the list which is different than having a trust line on the classic asset side.

**Pull request merged**